File: engine/core/base/lifecycle.cpp
#include "lifecycle.hpp"

namespace r2d2 {

Lifecycle::Lifecycle()
    : running_(false)
{}

void Lifecycle::init() {
    running_.store(true);
}

void Lifecycle::update() {
    // plus tard : update du monde, ECS, physics, etc.
}

void Lifecycle::shutdown() {
    running_.store(false);
}

uint8_t Lifecycle::is_running() const {
    return running_ ? 1 : 0;
}


} // namespace r2d2

File: engine/python/base/lifecycle.py
from engine.cython.base.lifecycle import PyLifecycle

class Lifecycle:
    def __init__(self):
        self._lc = PyLifecycle()

    def init(self):
        self._lc.init()

    def update(self):
        self._lc.update()

    def shutdown(self):
        self._lc.shutdown()

    def is_running(self):
        return self._lc.is_running()
